examples/webgl_buffergeometry_drawcalls_indexed.html

```diff
@@ -0,0 +1,297 @@
+				group.add( helper );
+
+
+				// create position attribute (for both pointCloud and lineMesh)
+				positions = new THREE.Float32BufferAttribute( maxParticleCount * 3, 3 ).setDynamic( true );
+				
+				for ( var i = 0; i < maxParticleCount; i++ ) {
+					
+					var x = Math.random() * r - r / 2;
+					var y = Math.random() * r - r / 2;
+					var z = Math.random() * r - r / 2;
+
+					// setXYZ() is usefull to set values. See documentation https://threejs.org/docs/#api/core/BufferAttribute
+					positions.setXYZ( i, x, y, z );
+					
+					particlesData.push( {
+						velocity: new THREE.Vector3( -1 + Math.random() * 2, -1 + Math.random() * 2, -1 + Math.random() * 2 ),
+						numConnections: 0
+					} );
+					
+				}
+
+				// create color attribute (only for lineMesh)
+				colors = new THREE.Float32BufferAttribute( maxParticleCount * 3, 3 );
+
+				// create index attribute (only for lineMesh)
+				var numIndices = maxParticleCount * ( maxParticleCount - 1 );
+				indices = new THREE.Uint32BufferAttribute( numIndices, 1 ).setDynamic( true );
+				
+				// create particles
+				var particleGeometry = new THREE.BufferGeometry();
+				particleGeometry.addAttribute( 'position', positions );
+				particleGeometry.setDrawRange( 0, particleCount );
+				
+				var particleMaterial = new THREE.PointsMaterial( {
+					color: 0xFFFFFF,
+					size: 3,
+					blending: THREE.AdditiveBlending,
+					transparent: true,
+					sizeAttenuation: false
+				});
+				
+				pointCloud = new THREE.Points( particleGeometry, particleMaterial );
+				group.add( pointCloud );
+				
+				// create lines
+				var lineGeometry = new THREE.BufferGeometry();
+				lineGeometry.addAttribute( 'position', positions );
+				lineGeometry.addAttribute( 'color', colors );
+				lineGeometry.setIndex( indices );
+				lineGeometry.computeBoundingSphere();
+				lineGeometry.setDrawRange( 0, 0 );
+
+				var lineMaterial = new THREE.LineBasicMaterial( {
+					vertexColors: THREE.VertexColors,
+					blending: THREE.AdditiveBlending,
+					transparent: true
+				} );
+
+				linesMesh = new THREE.LineSegments( lineGeometry, lineMaterial );
+				group.add( linesMesh );
+
+				renderer = new THREE.WebGLRenderer( { antialias: true } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+				renderer.gammaInput = true;
+				renderer.gammaOutput = true;
+
+				container.appendChild( renderer.domElement );
+
+				stats = new Stats();
+				container.appendChild( stats.dom );
+
+				window.addEventListener( 'resize', onWindowResize, false );
+
+			}
+
+			function onWindowResize() {
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+			}
+
+			function animate() {
+				
+				// update particle position
+				for ( var i = 0; i < particleCount; i++ ) {
+
+					var pData = particlesData[ i ];
```

Comment:
This must be initialized in its own loop as in the original code https://github.com/mrdoob/three.js/blob/dev/examples/webgl_buffergeometry_drawcalls.html#L209-L210
as you will be using querying `particleDataB` indexed by `j > i` without being initialized correctly. https://github.com/mrdoob/three.js/blob/dev/examples/webgl_buffergeometry_drawcalls.html#L237
https://github.com/mrdoob/three.js/blob/dev/examples/webgl_buffergeometry_drawcalls.html#L248